server/src/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -705,6 +744,30 @@ public boolean deleteVpcOffering(final long offId) {
     }
 
     @Override
+    public Map<String, String> updateQuaggaConfig(final Long zoneId, final String protocol, final String ospfArea, final Short helloInterval, final Short deadInterval,
+            final Short retransmitInterval, final Short transitDelay, final String authentication, final String quaggaPassword, final String superCIDR, final Boolean enabled)
+                    throws BadCIDRException {
+        OSPFZoneConfig qzc = new OSPFZoneConfig();
+        DataCenterVO dc = _dcDao.findById(zoneId);
+        _dcDao.loadDetails(dc);
+        Map<String, String> details = dc.getDetails();
+        qzc.setDefaultValues(details);
+        qzc.setValues(protocol, ospfArea, helloInterval, deadInterval, retransmitInterval, transitDelay, authentication, quaggaPassword, superCIDR, enabled);
+        details = qzc.getValues();
```

Comment:
Given the number of parameters on ``OSPFZoneConfig.setValues`` with the sequence of calls on lines 754-756, please consider extracting the creation of ``OSPFZoneConfig`` instances to a builder class as a static inner class on ``OSPFZoneConfig``.